### DISCUSSION TITLE
No documentation on how to use node packages in API functions

### DISCUSSION THREAD

**Discussion by jakeleventhal** [vercel/community#2375] [Help] [answered]
In my pages/api/my-route.ts file, I want to import `googleapis`, but it uses a bunch of node builtin modules, like http2. How can I work around this? It isn't listing in docs.

**Comment by amyegan** [comment, marked as answer]
Hi @jakeleventhal. It sounds like you're trying to use a package that requires Node.js or browser APIs in an Edge Function. One of the limitations is that [many Node.js APIs are not available in Edge Functions](https://vercel.com/docs/concepts/functions/edge-functions/limitations#many-node.js-apis-are-not-available). Do you get them same error when running it as a Serverless Function with the [Node.js Runtime](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js) instead of using the [Edge Runtime](https://vercel.com/docs/concepts/functions/edge-functions/edge-runtime)?

**Reply by jakeleventhal** [reply]
Yes, this is precisely my situation. I am trying to run a node.js function

**Reply by jakeleventhal** [reply]
I figured out the problem actually. But perhaps there is a better way

I have
```
src/
  api-utils/
    index.ts // just exports `GoogleUtils`
    GoogleUtils.ts // uses `googleapis`
  pages/
    api/
      some-route.ts // imports GoogleUtils
```

If i remove the api-utils/index.ts file, and just import `GoogleUtils` directly, there are no issues. The problem is that the index.ts file includes GoogleUtils in the browser bundle (this seems incorrect to me). If i am not actually importing GoogleUtils in any browser code, I'm not sure why this would happen.
